Cargo.toml

```diff
@@ -10,7 +10,8 @@ members = [
   "hir-type",
   "tools/*",
   "frontend-wasm",
-  "tests/rust-wasm-tests"
+  "compiler-tests-rust-source",
```

Comment:
*Nit*: I'd prefer to move these crates to the `tests` subdirectory, and give them simpler names, e.g. `tests/rust` and `tests/integration`; where the first is for testing compilation from Rust sources, and the latter is for integration tests of the compiler itself.

I'm open to suggestions as well, but would rather err on the side of broad strokes than getting too specific.

Reply:
Ha! Initially, I put them in `tests/compiler` and `tests/compiler-rust-source` but then I thought that since we use a "flat" folder structure for crates (e.g. `hir-*`), I should do the same for tests. 
I'll do `tests/integration` and `tests/rust`.

Reply:
In general the structure is flat like you expected, but I think of `midenc` and the `hir*` crates as the primary focus of the repo, and thus keep those at the root (though `midenc` needs to be moved to the root now that the repo is renamed). On the other hand, the test crates, and supporting tools like `filecheck` are auxiliary, so nesting those in folders like `test` and `tools` keeps the focus on the "public" face of the compiler, and avoids too much clutter in the root due to incidental things like what tools we're building as part of our infrastructure and that sort of thing.

With multi-crate projects like this, there are a number of different ways to organize them, and they all have their merits (e.g. `rust-lang/rust`, `cranelift`), so I'm not too fixated on this specific structure, but at least now you know my rationale 😄